src/zkp/musig.rs

```diff
@@ -0,0 +1,1518 @@
+    ) -> Result<PublicKey, MusigTweakErr> {
+        let cx = secp.ctx().as_ptr();
+        unsafe {
+            let mut out = PublicKey::from(ffi::PublicKey::new());
+            if ffi::secp256k1_musig_pubkey_xonly_tweak_add(
+                cx,
+                out.as_mut_c_ptr(),
+                self.as_mut_ptr(),
+                tweak.as_c_ptr(),
+            ) == 0
+            {
+                Err(MusigTweakErr::InvalidTweak)
+            } else {
+                Ok(out)
+            }
+        }
+    }
+
+    /// Starts a signing session by generating a nonce
+    ///
+    /// This function outputs a secret nonce that will be required for signing and a
+    /// corresponding public nonce that is intended to be sent to other signers.
+    ///
+    /// MuSig differs from regular Schnorr signing in that implementers _must_ take
+    /// special care to not reuse a nonce. If you cannot provide a `sec_key`, `session_id`
+    /// UNIFORMLY RANDOM AND KEPT SECRET (even from other signers).
+    /// Refer to libsecp256k1-zkp documentation for additional considerations.
+    ///
+    /// Musig2 nonces can be precomputed without knowing the aggregate public key, the message to sign.
+    /// See the `new_nonce_pair` method that allows generating [`MusigSecNonce`] and [`MusigPubNonce`]
+    /// with only the `session_id` field.
+    ///
+    /// Remember that nonce reuse will immediately leak the secret key!
+    ///
+    /// # Returns:
+    ///
+    /// A pair of ([`MusigSecNonce`], [`MusigPubNonce`]) that can be later used signing and aggregation
+    ///
+    /// # Arguments:
+    ///
+    /// * `secp` : [`Secp256k1`] context object initialized for signing
+    /// * `session_id`: [`MusigSessionId`] Uniform random identifier for this session. Each call to this
+    /// function must have a UNIQUE `session_id`.
+    /// * `pub_key`: [`PublicKey`] of the signer creating the nonce.
+    /// * `msg`: [`Message`] that will be signed later on.
+    /// * `extra_rand`: Additional randomness for mis-use resistance
+    ///
+    /// # Errors:
+    ///
+    /// * `ZeroSession`: if the `session_id` is supplied is all zeros.
+    ///
+    /// Example:
+    ///
+    /// ```rust
+    /// # # [cfg(any(test, feature = "rand-std"))] {
+    /// # use secp256k1_zkp::rand::{thread_rng, RngCore};
+    /// # use secp256k1_zkp::{MusigKeyAggCache, Secp256k1, SecretKey, Keypair, PublicKey, MusigSessionId, Message};
+    /// # let secp = Secp256k1::new();
+    /// # let sk1 = SecretKey::new(&mut thread_rng());
+    /// # let pub_key1 = PublicKey::from_secret_key(&secp, &sk1);
+    /// # let sk2 = SecretKey::new(&mut thread_rng());
+    /// # let pub_key2 = PublicKey::from_secret_key(&secp, &sk2);
+    /// #
+    /// let key_agg_cache = MusigKeyAggCache::new(&secp, &[pub_key1, pub_key2]);
+    /// // The session id must be sampled at random. Read documentation for more details.
+    /// let session_id = MusigSessionId::new(&mut thread_rng());
```

Comment:
This method doesn't exist anymore and seems to be renamed to `assume_unique_per_nonce_gen`.